src/components/Map/EditableMap/EditableMap.js

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import MapGL from 'react-map-gl'
+import { Editor, DrawPolygonMode, EditingMode } from 'react-map-gl-draw'
+import Icon from '../../Icon'
+import { mapbox } from 'config'
+
+import { getEditHandleStyle, getFeatureStyle } from './EditableMapStyles'
+import './EditableMap.scss'
+
+export default function EditableMap (props) {
+  const { locationObject, polygon, savePolygon } = props
+  const centerAt = locationObject?.center
+  const showLocationOnMap = !polygon && centerAt
+  const [viewport, setViewport] = useState(null)
+  const [mode, setMode] = useState(null)
+  const [selectedFeatureIndex, setSelectedFeatureIndex] = useState(null)
+  const [isModeDrawing, setIsModeDrawing] = useState(false)
+  const editorRef = useRef(null)
+  const existingPolygon = polygon ? [{
+    geometry: polygon,
+    properties: { },
+    type: 'Feature'
+  }] : [{
+    geometry: { },
+    properties: { },
+    type: 'Feature'
+  }]
+
+  useEffect(() => {
+    const defaultLocation = showLocationOnMap ? {
```

Comment:
shouldnt this location be based on the polygon if that exists first?

Reply:
This is getting location based on pin/precise location, not polygon location. It centers the map on `defaultLocation` which defaults to Oakland if no `locationObject` exists. 

in the future it would be great to find the center of the polygon location using a bounding box